setup_template.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin python3
+
+import os
+import pathlib as pl
+
+
+def main():
```

Comment:
For ease of future maintenance, I'd split this up into a few functions and have `main()` be the orchestrator. My suggestions would be: 
- `parse_git_metadata() -> dict[str, str]` 
- `query_user_for_repo_settings(git_metadata) -> dict[str, str]`
- `replace_text_in_files(files: list[pl.Path], replacements: dict[str, str]) -> None`

Reply:
Generally I of course agree, but as this is meant to be used as a script with a potentially very low version system python (so no type hints, dataclasses, etc. to clean it up) I think as long as its only around 100 lines this does more harm than good